contracts/staking/tests/NativeTokenStakingManagerTests.t.sol

```diff
@@ -0,0 +1,324 @@
+            })
+        );
+        vm.mockCall(
+            WARP_PRECOMPILE_ADDRESS,
+            abi.encode(IWarpMessenger.sendWarpMessage.selector),
+            abi.encode(bytes32(0))
+        );
+        vm.expectCall(
+            WARP_PRECOMPILE_ADDRESS,
+            abi.encodeCall(IWarpMessenger.sendWarpMessage, registerSubnetValidatorMessage)
+        );
+        app.resendRegisterValidatorMessage(validationID);
+    }
+
+    function testCompleteValidatorRegistration() public {
+        _setUpCompleteValidatorRegistration({
+            nodeID: DEFAULT_NODE_ID,
+            subnetID: DEFAULT_SUBNET_ID,
+            weight: 1e6,
+            registrationExpiry: DEFAULT_EXPIRY,
+            signature: DEFAULT_ED25519_SIGNATURE,
+            registrationTimestamp: 1000
+        });
+    }
+
+    function testInitializeEndValidation() public {
+        _setUpInitializeEndValidation({
+            nodeID: DEFAULT_NODE_ID,
+            subnetID: DEFAULT_SUBNET_ID,
+            weight: 1e6,
+            registrationExpiry: DEFAULT_EXPIRY,
+            signature: DEFAULT_ED25519_SIGNATURE,
+            registrationTimestamp: 1000,
+            completionTimestamp: 2000
+        });
+    }
+
+    function testInitializeEndValidationWithUptimeProof() public {
+        // TODO: implement
+    }
+
+    function testInitializeEndValidationExcessiveChurn() public {
+        // TODO: implement
+    }
+
+    function testResendEndValidation() public {
+        bytes32 validationID = _setUpInitializeEndValidation({
+            nodeID: DEFAULT_NODE_ID,
+            subnetID: DEFAULT_SUBNET_ID,
+            weight: 1e6,
+            registrationExpiry: DEFAULT_EXPIRY,
+            signature: DEFAULT_ED25519_SIGNATURE,
+            registrationTimestamp: 1000,
+            completionTimestamp: 2000
+        });
+        bytes memory setValidatorWeightPayload =
+            StakingMessages.packSetSubnetValidatorWeightMessage(validationID, 0, 0);
+        vm.mockCall(
+            WARP_PRECOMPILE_ADDRESS,
+            abi.encode(IWarpMessenger.sendWarpMessage.selector),
+            abi.encode(bytes32(0))
+        );
+        vm.expectCall(
+            WARP_PRECOMPILE_ADDRESS,
+            abi.encodeCall(IWarpMessenger.sendWarpMessage, setValidatorWeightPayload)
+        );
+        app.resendEndValidatorMessage(validationID);
+    }
+
+    function testCompleteEndValidation() public {
+        bytes32 validationID = _setUpInitializeEndValidation({
+            nodeID: DEFAULT_NODE_ID,
+            subnetID: DEFAULT_SUBNET_ID,
+            weight: 1e6,
+            registrationExpiry: DEFAULT_EXPIRY,
+            signature: DEFAULT_ED25519_SIGNATURE,
+            registrationTimestamp: 1000,
+            completionTimestamp: 2000
+        });
+
+        bytes memory subnetValidatorRegistrationMessage =
+            StakingMessages.packSubnetValidatorRegistrationMessage(validationID, false);
+        vm.mockCall(
```

Comment:
these `mockCall` and `expectCall`s on the Warp precompile will likely be reused throughout, can we create helper functions for them?

Reply:
Good call, added helpers in